### PR TITLE
chore(file-download): enable downloading folder/file contents from file directory

### PR DESCRIPTION
**Pull request by priyakanabar-crest**
This PR adds the ability to download the contents of a folder/file directly from the file directory.

### CODE REVIEW

File: zt_backend/router.py [hunk 1]
```diff
@@ -588,7 +594,135 @@ def delete_item(delete_request: request.DeleteItemRequest):
...
+        '.7z': 'application/x-7z-compressed',
+        '.rar': 'application/vnd.rar',
+        '.webp': 'image/webp',
+        # Add more custom MIME types here as needed
+    }
+
+    for ext, mime_type in custom_mime_types.items():
+        mimetypes.add_type(mime_type, ext)
+
+def get_mime_type(filename: str) -> str:
```
Comment (Carson-Shaar) [inline zt_backend/router.py:615]: Lets break this all out into a utils file. In the same vein any helper functions for file uploads/changes should be moved there. We want to keep the router as only endpoints as much as possible

File: zt_frontend/src/components/FileFolderDownloadDialog.vue [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<template>
+    <div>
```
Comment (Carson-Shaar) [inline zt_frontend/src/components/FileFolderDownloadDialog.vue:2]: All wrapped in a div?

**Comment by Carson-Shaar** [comment]
Will want to test this in dev as well

File: zt_backend/router.py (hunk 1 above)
Comment (priyakanabar-crest) [reply zt_backend/router.py:615]: okay , I will update this

File: zt_frontend/src/components/FileFolderDownloadDialog.vue (hunk 2 above)
Comment (priyakanabar-crest) [reply zt_frontend/src/components/FileFolderDownloadDialog.vue:2]: let me check on this

**Comment by priyakanabar-crest** [comment]
@Carson-Shaar I have updated the code and rebase the branch

File: setup.cfg [hunk 3]
```diff
@@ -32,6 +32,7 @@ install_requires =
     setuptools<=68.2.2
     pygwalker<=0.4.8.1
     openai<=1.1.0
+    aiofiles<=24.1.0
```
Comment (Carson-Shaar) [inline setup.cfg:35]: Are we using this?

File: zt_frontend/src/components/FileExplorer.vue [hunk 4]
```diff
@@ -79,6 +87,12 @@
     </template>
   </v-snackbar>
 
+  <FileFolderDownloadDialog
```
Comment (Carson-Shaar) [inline zt_frontend/src/components/FileExplorer.vue:90]: All of the dialogs should live under the list item like this:

            <v-list-item
              v-if="<some_bool>"
              link
              prepend-icon="mdi-account-outline"
              title="Manage Access"
            >
              <ShareProjectDialog :project-name="item.name" />
            </v-list-item>
            
   Note the link directive

**Comment by Carson-Shaar** [comment]
Please rebase this to resolve conflicts and test with very large folders

**Comment by priyakanabar-crest** [comment]
Rebased solve conflicts and updated downlader dialog to be in v-list-item

**Review by Carson-Shaar — APPROVED** [review]

**Pull request merged**